utils/logger/logger_test.go

```diff
@@ -0,0 +1,348 @@
+// this is a new logger interface for mattermost
+
+package logger
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+)
+
+type LogMessage struct {
+	Context map[string]string
+	File    string
+	Message string
+}
+
+// ensures that values can be recorded on a Context object, and that the data in question is serialized as a part of the log message
+func TestSerializeContext(t *testing.T) {
+	t.Run("Context values test", func(t *testing.T) {
```

Comment:
I don't think there's any reason to put all of the tests inside `t.Run` calls (`t.Run` is just meant for sub-tests like when we use table-driven tests).